app/src/processing/app/contrib/ContributionManager.java

```diff
@@ -284,13 +287,118 @@ public boolean accept(File dir, String file) {
     } catch (IOException e) {
 //      Again, forget about the stack trace. The user ain't done wrong
 //      e.printStackTrace();
-      System.err.println("The unupdated contribution marker seems to not like "
-        + ac.getName() + ". You may have to install it manually to update...");
+      System.err.println(Language
+        .interpolate("contrib.startup.errors.new_marker", ac.getName()));
     }
 
   }
 
 
+  /**
+   * Blocking call to download and install a set of libraries. Used when a list
+   * of libraries have to be installed while forcing the user to not modify
+   * anything and providing feedback via the console status area, such as when
+   * the user tries to run a sketch that imports uninstaled libraries.
+   *
+   * @param aList
+   *          The list of AvailableContributions to be downloaded and installed.
+   */
+  public static void downloadAndInstallOnImport(final Base base,
+                                                final ArrayList<AvailableContribution> aList) {
+
+    // To avoid the user from modifying stuff, since this function is only called
+    // during pre-processing
+    base.getActiveEditor().getTextArea().setEditable(false);
+    base.getActiveEditor().getConsole().clear();
+
+    ArrayList<String> installedLibList = new ArrayList<String>();
+
+    // boolean variable to check if previous lib was installed successfully, 
+    // to give the user an idea about progress being made.
+    boolean isPrevDone = false;
+
+    for (AvailableContribution ad : aList) {
+      if (ad.getType() != ContributionType.LIBRARY)
+        continue;
+      try {
+        URL url = new URL(ad.link);
+        String filename = url.getFile();
+        filename = filename.substring(filename.lastIndexOf('/') + 1);
+        try {
+
+          File contribZip = File.createTempFile("download", filename);
+          contribZip.setWritable(true);
+
+          try {
+            // Use the console to let the user know what's happening
+            // The slightly complex if-else is required to let the user know when
+            // one install is completed and the next download has begun without 
+            // interfereing with occur status messages that may arise in the meanwhile
+            String statusMsg = base.getActiveEditor().getStatusMessage();
+            if (isPrevDone)
+              base
+                .getActiveEditor()
```

Comment:
We don't use this type of indenting elsewhere in the source code. Just get things on one line, or if they don't fit in 80 chars, break up the statements.